### ISSUE TITLE
Failure to start on Android 8.0 Oreo

### ISSUE THREAD

**Issue by gnomon-** [corbinlc/GNURootDebian#206]
When I launch the application on my Pixel XL running Android 8.0.0, the window animation displays then immediately disappears, leaving a translucent "Window 1" banner for a second or two.

Running kernel 3.18.52-g06de20685970.  I installed GNURoot Debian prior to the upgrade to 8.0.0.

**Comment by drhuffman12** [comment]
Would un-installing and re-installing GNURoot help?

**Comment by kenyoung** [comment]
I have the same issue on a Google Pixel XL phone running exactly the same kernel and Android release as gnomon mentioned above.   I have uninstalled and re-installed GNURoot Debian several times.   I get the message "GNURoot not yet installed. This could take a bit. Please wait." on a translucent window which disappears after ~1 second, and is followed by the "Window 1" banner that gnomon mentioned.

**Comment by BearChavez** [comment]
I have the same issue

**Comment by yochananmarqos** [comment]
Same here on my Nexus 5X. Logcat attached.
[2017-09-14-17-56-39.txt](https://github.com/corbinlc/GNURootDebian/files/1304909/2017-09-14-17-56-39.txt)

**Comment by rcollins0618** [comment]
Same issue here (exactly as kenyoung described).
Android 8.0.0,
Security patch level September 5, 2017
Kernel 3.18.52

**Comment by rcollins0618** [comment]
Forgot to mention: Google Pixel XL

**Comment by ifuller1** [comment]
I'm also seeing this.

**Comment by ifuller1** [comment]
Same Android device and software as above.

**Comment by tuomas2** [comment]
Same here with Xperia XZ1 Compact.

**Comment by mhofert** [comment]
Same here on Nexus 6P

**Comment by grettke** [comment]
Today I ran into the same issue described by @gnomon- on the same hardware and OS as @gnomon-

**Comment by cbeley** [comment]
Same here on Pixel C.

**Comment by bluelightning32** [comment]
It's failing due to the seccomp filter introduced in Android Oreo.

https://android-developers.googleblog.com/2017/07/seccomp-filter-in-android-o.html

**Comment by etherealmccoy** [comment]
This issue/thread may be relevant: 
  https://github.com/proot-me/PRoot/issues/106

**Comment by corbinlc** [comment]
I am back from the dead for a little while.  I am looking into this issue as my highest priority with the time that I have.  It is related to seccomp filtering in Oreo (good job to those that figured that out on their own).  The specific issue is that the version of busybox I am using was compiled against glibc and not bionic and some combination of that and what the busybox ash interpreter code is doing is calling one of the now blocked system calls (swapon/swapoff are the most likely candidates, but they block a few more).

**Comment by corbinlc** [comment]
If there are other illegal system calls being used by user apps, I can get around most of them by changing proot itself, but got to rebuild busybox first.

**Comment by etherealmccoy** [comment]
Very good news. Thanks Corbin. Happy to do some testing.

**Comment by gnomon-** [comment]
@corbinlc brilliant, thank you so much for the informative update.  Really glad to gain more understanding of the root cause.

**Comment by corbinlc** [comment]
Specifically, my old build of Busybox was calling the sigprocmask system call which is not an allowed system call on Oreo.  Compiling a static Busybox with the latest bionic makes it call rt_sigprocmask instead and that is allowed.  Need to test if this is the only issue or just the first issue.

**Comment by Yoone** [comment]
Any news on this? I am getting the behavior on my Google Pixel XL with Android 8.0.0.

**Comment by joelyoungblom** [comment]
I am experiencing the same issue on my Android 8.1.0 Pixel XL

**Comment by JonasFranzDEV** [comment]
Same issue on my OnePlus 3T with Android 8.0.

**Comment by pbreneman** [comment]
Same issue on Moto Z2 with Android 8.0 . I'm hoping to get this (popular) page working again:
http://www.turbocontrol.com/gnuroot.htm

**Comment by rrodrigueznt** [comment]
Same issue on my Xperia XZ running on Android 8.0.0.

Thanks for your work!!!

**Comment by tedb** [comment]
I am also seeing this on a Nexus Player with Android 8.0.0.

**Comment by Agent471** [comment]
Same issue on OP3 running Android 8.0

**Comment by xwct** [comment]
happens on my pixel xl with 8.1 jan security update

**Comment by monsterkali666** [comment]
same problem here
xiaomi mix 2 - 8.0

**Comment by Basti922** [comment]
Same issue

**Comment by oganm** [comment]
Hi everyone, this issue exists on all android build >8.0. At this point you should post if you can make this work rather than pointing out that it doesn't. I have not relation to the development of this project but some of us are watching this thread to be notified when there are news related to the issue and "me too" comments are distracting to those of us who use github on a daily basis. 

If you are effected by this issue and want to let the devs know that you exist, you can react to an existing post that voices your opinions with a thumbs up using the button below, located above each post:
![image](https://user-images.githubusercontent.com/6352379/36085770-266201a0-0f7d-11e8-8847-0b17d0b5cb99.png)

If you want to be notified of updates you can subscribe to the topic using the button below, located to the left of this window:
![image](https://user-images.githubusercontent.com/6352379/36085804-5b012bac-0f7d-11e8-9916-9eb5d078ee9e.png)

**Comment by corbinlc** [comment]
Sorry for such a long delay on this.  This problem is fully understood, but work is 1) a little bigger than expected and 2) this project is no long being supported by my employer, such that I don't have much time to work through it.  All that said, I have 1) now rebuilt all of the executables that are in the support directory using bionic such that they no longer use system calls that are not allowed 2) made it so SIGSYS signals are handled in an appropriate way and don't just cause the whole thing to crash and 3) started a new PRoot extension which maps now illegal system calls to ones that are legal.  I am not done with 3), but don't believe there is a lot of work yet.  I have a few things running inside the PRooted system on 8.0+, but not enough to release to anyone yet.  I am not going to give a specific date for when this will be finished, as I have limited time, but it won't take many working hours to get there at this point.

On a separate note, there is a good chance much more developer effort will be put into this soon fixing this issue if it is not already and taking this so much further.  I know this project has seemed dead for a while, but the news of its demise, by certain people, was premature.

**Comment by Root-voidX** [comment]
Any updates? I am hoping a beta build will come out before the s9 retail launch 🤞

**Comment by corbinlc** [comment]
Ok.  I haven't been able to work on this more since my last update, but we do have a new sponsor (more details on this soon).  There will be a paid full time developer starting at the end of April to move this project forward, as well as additional resources thereafter.  In addition, I will be spending much more time on this.  Combined we will make this project better than it has ever been and solve all the major current issues as well as introduce you all to a much better version.  The timeline isn't exact yet, but 8.0 issues should be solved in the next 2 months.  We will be more open about our upcoming features and release plan once we get up and running.

**Comment by marksteward** [comment]
In case it helps, the showstopper Oreo issues have been worked around in termux-proot with https://github.com/termux/proot/commit/454b0b1. You might be able to backport it.

I really like the way this works out-of-the-box, but I wonder whether it's better to port this to be a termux package? That way, the terminal and build chain comes free.

**Comment by corbinlc** [comment]
@marksteward Thanks.  I have seen what the termux team did here and it is very helpful.  Their solution solves some major issues caused by 8.0, but not all of them.  I do build the executables that run outside of proot like them now, but I have more proot extension than they do which need a little more rework and then I need to correctly handle some system calls that the termux team's work does not cover.  Both of those have started.  

I currently don't plan to make this a termux package, but I do plan to make it so termux, connectbot or other apps can easily connect to this out of the box so they can work together.

**Comment by gund** [comment]
I just want to mention that I'm using termux currently and even though it does work - it has some problems.

In particular when I install ubuntu under it - some packages will fail on installation (like LXDE) with some permission errors (probably because of non rooted device).
But I hope that it should be possible to run graphical environment on non-rooted device, so would like to help testing this app on my Pixel 2 XL.

**Comment by Root-voidX** [comment]
I assume the x integration is different in gnuroot? Though yes having termux shell built in on the next release would be sweet along with aarch64 support and a way to go around selinux. 

I have been using termux-arch and have had a fluid time. Lxde works and I have octave and network support. Still haven't figured out how to get acceleration working but termux is a pretty good alternative for now. 

It does seem termux-fedora lost dev support and termux-ubuntu is shit.

**Comment by gund** [comment]
@Root-voidX I have used termux-arch and installed LXDE but it was not able to connet to display: upon running `startlxde` getting error `Gtk-WARNING **: cannot open display: 0` (was trying to use it with `XServer XSDL app`).

So I thought will try different linux distros but so far no luck.
And since none of them allow me to run GUI I stick with Ubuntu because of it's massive apt repo.

P.S. Did you do any specific trick to launch lxde on your device and is it rooted?

**Comment by Root-voidX** [comment]
@gund

Open xsdl, switch back to termux and type in:

DISPLAY=:0 startlxde and then switch back to xsdl, it should start. I still haven't figured out hardware acceleration yet. At least with debt based distro installing libgl_mesa_swx11 works but haven't found an equivalent in arch.

**Comment by gund** [comment]
@Root-voidX yes I did that stuff but it still was complaining and I have no idea why.

I was able to only run GUI in plain termux but it is not much usefull anyway...

**Comment by Root-voidX** [comment]
@gund

Interesting, I guess you gotta be fast? Works fine on my s7e on 7.0 and my S9+ on 8.0

**Comment by gund** [comment]
@Root-voidX do not know what you mean by `gotta be fast` =)

Anyway I'm tried to run that on my Pixel 2 XL non-rooted and getting that error.
Googling exactly that erro message literally did not give me any usefull info.

**Comment by corbinlc** [comment]
Guys, I am going to do my best to make sure this all works really well in GNURoot Debian and what comes after.  With regards to how this all behaves in termux, please file an issue against termux and discuss those details there.  That would be most helpful to them.

**Comment by Root-voidX** [comment]
What I mean is you have to quick in switching apps as the error means that termux doesn't find a running Xserver to export a display. That's why I said if you switch back to XSDL fast enough it should work. 😀

**Comment by pantumaca** [comment]
@gund Try this:

export DISPLAY=localhost:0
export PULSE_SERVER=localhost:4712
start"desktopyouwant"

It was making me crazy until i tried that... and working with lxde, lxqt and bungie in android 8.0 in a HTC U11

Really nice this tarmux app

**Comment by gund** [comment]
Wow @pantumaca you are genius!

It worked! Dead simple!..

Where have you been all this time? =)

Thanks a lot for sharing here👍

**Comment by pantumaca** [comment]
@gund Really? It worked? I am really happy to be useful!

I must be honest, that info come from https://github.com/xeffyr/termux-x-repository

Now time to enjoy! jejeje

**Comment by Root-voidX** [comment]
Uh, does gnuroot debian work on 8.0 @pantumaca?

Also you can just type in DISPLAY=:0 which means localhost anyway so idk why that didn't work but eh :/

**Comment by pantumaca** [comment]
@Root-voidX  Nope... It says "Not installed yet" when you try to launch the app. I hope we can see some update in June as the dev said.

Yeah, DISPLAY=:0 shoud work but there must be something, I do not know if it is related to Termux or android8.... who knows... Anyway, if that command does not work then use DISPLAY=localhost:0.

**Comment by gund** [comment]
Btw I just checked and it is purely DISPLAY variable that needs to have localhost.
PULSE_SERVER can be set with tcp127.0.0.1 without any problems

**Comment by salvomic** [comment]
I've too the same problem in Huawei Mediapad M5 (Android 8, Oreo): "GNURoot not yet installed... This could take a bit. Please wait"... The app doesn't open.

**Comment by Skrmnghrd** [comment]
Hello, running on xperia xz1 oreo too. Had this problem :(. Calling an illegal syscalls? maybe?

**Comment by salvomic** [comment]
@Skrmnghrd: the problem should be related to the seccomp filter in oreo (see above @bluelightning32 post)... :-(

**Comment by corbinlc** [comment]
Update: As of yesterday, there are 2 developers working on this project again (this time they are paid).  Fixing 8.0 will be in the first release we make.

**Comment by salvomic** [comment]
well, thank you!

**Comment by Skrmnghrd** [comment]
om my god. that is awesome :) Its such  a good news. It really is spring :))

**Comment by Root-voidX** [comment]
@corbinlc 

This is great news. I would like to add that the proot method based of termux/proot failed on newer devices running the latest kernals (i.e. SD 845 devices such as Samsung Galaxy S9)

https://github.com/termux/proot/issues/13#issuecomment-381816080

This the github I was helping with. I only wanted to bring this your attention since if this really is a proot implementation issue with the newer kernal, sooner or later the gunroot team will be exposed to this issue as well!

**Comment by salvomic** [comment]
please, include also Huawei Mediapad M5 8" with proot facility...

**Comment by AdamMatthewLTM** [comment]
Hi all,

I am currently assisting the developers in sorting relevant issues for GNURoot Debian on GitHub. If the issue discussed is still relevant, could you please reply to this thread and state the issue needs to be resolved and I will attempt to sort it for you ASAP. If the issue is no longer relevant, please reply to this thread and state it has been resolved, and I will suggest this thread be closed.

Furthermore, if a particular issue is considered to be high priority, please state that you are requesting it to be considered as high priority and I will mark it as such.

Thank you all so much for your time and understanding,
-Adam

**Comment by rootex-** [comment]
@AdamMatthewLTM yes, it's still relevant

**Comment by salvomic** [comment]
yes, still relevant and severe: it doesn’t install in 8.0 (oreo).

Salvo

**Comment by romeupalos** [comment]
Yes! Still relevant and top priority since it's a showstopper for some
people.

Thanks,
Romeu

Em sex, 11 de mai de 2018 03:42, salvomic <notifications@github.com>
escreveu:

**Comment by gitawego** [comment]
yes, the most critical bug imo.

**Comment by corbinlc** [comment]
Yes.  Definitely #1 issue.

**Comment by corbinlc** [comment]
BTW.. I now have this fixed locally.  Release of this plus more in about
1month.  Maybe a beta to try sooner.

**Comment by salvomic** [comment]
thanks.
Without that upgrade those of us have Oreo cannot help to beta testing…

Salvo

> Il giorno 11 mag 2018, alle ore 20:09, corbinlc <notifications@github.com> ha scritto:
> 
> BTW.. I now have this fixed locally. Release of this plus more in about
> 1month. Maybe a beta to try sooner.
>

**Comment by ksaves2** [comment]
Get Oreo fixed and a lot more folks can get back onboard.  Get a work around for attached B/T devices 
and that will allow a lot more utility for those who want to use B/T attached devices with our utilities.
So far /dev/rfcomm"X" is invisible to GNURoot Debian/LXDE.  Even if there was something like 
"Share GPS" that uses gpsd to share the onboard GPS with the GRD side works and if a Bluetooth utility on the "Android side" could make an attached peripheral (outside of a mouse and keyboard that are already there if paired) visible to the GRD "side" that would be workable.  I believe no other "Linux on Android" can do this trick without some additional code.  Some have posted they've done it but the code is pretty rough.  (I saw that on the Linux Deploy thread.)  Kurt

**Comment by augesrob** [comment]
Samsung galaxy s9 plus not installed yet issue still occurs

**Comment by corbinlc** [comment]
As mentioned, we have this fixed locally and will release around June 15th
(we are changing a lot of things).  We may do a public beta before that.
This is a pretty nasty issue so I expect that there will be some corner
cases that the beta testers would be able to uncover.

**Comment by Root-voidX** [comment]
Cant wait to beta test. I suspect there will be some issues with devices with newer kernels such as SD 845 devices. Also ksaves2 is partly wrong, termux proot works on 8.0, currently running arch. For other peripherals its already possible with chroot and mount points. Idk if proot is possible with the permission issue.

**Comment by AdamMatthewLTM** [comment]
All, 

Thank you all so much for your responses, and my sincerest apologies for the delay. I am working backwards from oldest to newest issues on github, and have marked this one as the **-highest-** priority to work on. 

I will be recommending that, **once a beta release for Oreo is released**, new threads be created specifically for debugging issues with the beta release and that this thread be recommended as closed.

Thank you all for your time and patience,
-Adam

**Comment by diedburn** [comment]
Any progress on this?

**Comment by ksaves2** [comment]
I don't think an update has been posted.  Kurt

**Comment by lapsio** [comment]
Is it there yet? Today I updated to Oreo without knowing that it's gonna turn my phone into Nokia 3310 functionality wise...

**Comment by marianofxp** [comment]
Any update ? Android 8

**Comment by corbinlc** [comment]
We have started an internal beta right now.  We will have a public beta in about 1 week.  I will post information on how to join that then.  Sorry I have been quiet but I have been very busy developing what comes next.

**Comment by corbinlc** [comment]
This issue is specifically about an issue that only occurs in 8.0+.  Please file a new issue

**Comment by DeicPro** [comment]
@corbinlc great, I can't wait to test that beta! :D. I miss play RuneScape from the smartphone.

**Comment by ricecube** [comment]
@corbinlc can't wait to test that beta!

**Comment by corbinlc** [comment]
All, the beta is up.  You need to sign up as a tester here:
https://play.google.com/apps/testing/tech.ula

A few things:
This is a new app called UserLAnd, not GNURoot Debian, but similar.  It already does what the core GNURoot Debian did, but better.
Right now you can setup a Debian filesystem only, but other distros and specific applications will be provided soon.
Right now you can have it run an ssh or vnc server within that filesystem and then automatically connect to it using ConnectBot or bVNC.  More choices will be coming here too.
It takes a several minutes to start your session the first time.

Please start beating on it and filing issues against that github repo.

**Comment by corbinlc** [comment]
Oh, it is still free as in speech and free as in beer.

**Comment by oganm** [comment]
was testing this and manage to miss the default password message and can't bring it back... so... what was the default password again?

**Comment by corbinlc** [comment]
userland

**Comment by ricecube** [comment]
The default pass is userland , the same name as the app :)

**Comment by pantumaca** [comment]
Termius Ssh client is working out of the box.

i had installed it because i use xrdp over ssh to connect to my home pc (elementary).

thank you so much for this beta!!

**Comment by LordButtWiggle** [comment]
Is there a way to access the phone storage through the linux distro? Kind of like Termux does with setup-external-storage command

**Comment by Root-voidX** [comment]
I was able to launch without issues, just I cannot use apt-get. The error it gives me is this:


The programs included with the Debian GNU/Linux system are free sof
ware;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
user@localhost:~$ apt-get update
Reading package lists... Error!
E: Could not open lock file /var/lib/apt/lists/lock - open (13: Per
ission denied)
E: Unable to lock directory /var/lib/apt/lists/
E: Could not create temporary file for /var/cache/apt/srcpkgcache.b
n - mkstemp (13: Permission denied)
E: The package lists or status file could not be parsed or opened.
user@localhost:~$

**Comment by keaneokelley** [comment]
Use the new git repo for the UserLAnd app https://github.com/CypherpunkArmory/UserLAnd/

**Comment by ricecube** [comment]
@root-voidx try entering sudo su to get to fake root first and see if that works

**Comment by Root-voidX** [comment]
Thank you guys, using sudo works. Thats cool!

**Comment by marianofxp** [comment]
(lxsession:32685): Gtk-WARNING ++: cannot open display: :0

**Comment by ricecube** [comment]
@root-voidx if you already have installed lxde and have xserver downloaded this command should enable the mounted lxde. 
DISPLAY=:0 PULSE_SERVER=tcp:127.0.
.1:4712

startlxde &

**Comment by rodrigovilaca** [comment]
How does access the phone storage?

**Comment by oganm** [comment]
https://github.com/CypherpunkArmory/UserLAnd/issues/46

**Comment by ricecube** [comment]
@sciipking I've been trying to get osrs running, it's much more challenging to do then with gnuroot, you have to install Java jre 8 then the hakariknight rs repository, and pull the install from your internal storage pathway which is different for each phone vendor. If you get it working let me know, I haven't had the time recently to finish up.

**Comment by corbinlc** [comment]
We are going to automate this in the future, but getting the more generic
case working well first.

**Comment by ricecube** [comment]
@corbinlc the shell has been working great for me so far, got lxde installed and running fine through xserver! I'll keep playing around with it and post any issues I find!

**Comment by corbinlc** [comment]
Feel free to contribute interesting screen shots for our GitHub repo and so
we can use it in our store listing.

**Comment by DeicPro** [comment]
@ricecube I didn't got run in xserver xsdl. I installed unix-runescape-client, openbox and openjdk-jre-8

**Comment by DeicPro** [comment]
Ok I got it reading this: https://github.com/CypherpunkArmory/UserLAnd/issues/45

But when I run Runescape it send me some error by perl module dependencies and crash. I will try to manual install those modules after

**Comment by snoopy221** [comment]
Anyone happen to have the runescape script to run on userland?

**Comment by DeicPro** [comment]
I don't got that Java works. I tried Oracle's JDK 7u60 (java not found) and OpenJDK JRE 8 which gives me some Java errors and RuneScape crash before loading the client.

**Comment by DeicPro** [comment]
Seems the problem is that Oracle JDK need /lib/ld-linux.so.3, I tried simlink existing ld-linux* libs, but didn't work:
```
user@localhost:~$ java --version                                    
-bash: /usr/bin/java: No such file or directory           
              
user@localhost:~$ strings /usr/bin/java                   
/lib/ld-linux.so.3                                        
...

user@localhost:~$ ls /lib/ld*                             
/lib/ld-linux-aarch64.so.1  /lib/ld-linux.so.2
                                   
user@localhost:~$ sudo ln -s /lib/ld-linux.so.2 /lib/ld-linux.so.3                                      
                                         
user@localhost:~$ java --version                          
Segmentation fault                
                        
user@localhost:~$ sudo rm /lib/ld-linux.so.3   
           
user@localhost:~$ sudo ln -s /lib/ld-linux-aarch64.so.1 /lib/ld-linux.so.3   
                                    
user@localhost:~$ java --version                          
Inconsistency detected by ld.so: rtld.c: 1282: dl_main:
 Assertion `GL(dl_rtld_map).l_libname' failed!
```

**Comment by pantumaca** [comment]
I am not sure if this is the right place... Have anyone running firefox or libreoffice?

with firefox i get a blank screen (tried the xrender option in about:config with true and false)
Libreoffice returns "no proc mounted"

i also tried others browsers as chromium (cannot create new namespace) and epiphany (i get a loop with an error i cannot remember).

**Comment by Skrmnghrd** [comment]
Yes. I tried firefox. Its just blank. (It loads but it dosent display. ) Im
running lxde. I installed lxde like anormal one and made it run via exec
startlxde

On 25 Jun 2018 1:47 PM, "pantumaca" <notifications@github.com> wrote:

I am not sure if this is the right place... Have anyone running firefox or
libreoffice?

with firefox i get a blank screen (tried the xrender option in about:config
with true and false)
Libreoffice returns "no proc mounted"

i also tried others browsers as chromium (cannot create new namespace) and
epiphany (i get a loop with an error i cannot remember).

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/corbinlc/GNURootDebian/issues/206#issuecomment-400055569>,
or mute the thread

**Comment by Root-voidX** [comment]
You cant run any program that requires rendering without hardware/software rendering. A fix was to run libgl1-mesa-swx11, but i cant seem to find it in debian stretch. If you get that, the system will software render for you.

**Comment by DeicPro** [comment]
```
user@localhost:~$ sudo apt-file search ld-linux.so.3   
libc6-armel-cross: /usr/arm-linux-gnueabi/lib/ld-linux.so.3                                                   
libc6-armhf-cross: /usr/arm-linux-gnueabihf/lib/ld-linux.so.3
```

I tried symlink both to /lib/ld-linux.so.3, but segmentation fault in both

**Comment by DeicPro** [comment]
@Root-voidX both obviously.

In the last two messages was with soft float version of Oracle JDK. Now i'm trying with Hard Flost again and looks left arm 32 bits libraries in the system (and a proper interpreter), but I don't got the solution.

```
user@localhost:~$ file /usr/bin/java
/usr/bin/java: symbolic link to /etc/alternatives/java                                                          
                                                                                                                                          
user@localhost:~$ file /etc/alternatives/java    
/etc/alternatives/java: symbolic link to /opt/jdk
1.7.0_60/jre/bin/java                                                     

user@localhost:~$ file /opt/jdk1.7.0_60/jre/bin/j
ava
/opt/jdk1.7.0_60/jre/bin/java: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, for GNU/Linux 2.6.26, BuildID[sha1]=217d6afa01cd20597fde7298fba1268ca6d08500, not stripped          
```

Too `ldd` shows this in all symbolic link & original java path:
```
not a dynamic executable
```

**Comment by Root-voidX** [comment]
@DeicPro. Weird i run Java no issue on s9+. Did install everything related to java? Search java using sudo apt-cache search java and install all packages?

**Comment by DeicPro** [comment]
@Root-voidX I want to install Oracle JDK 7u60 (due the people says gives better performance to RuneScape) and it is installed manually, but it is 32bit armel/armhf.

I tried:
```
sudo dpkg --add-architecture armhf
sudo apt update
sudo apt install libc6:armhf
```

But `Segmentation fault` when call `java`

So IDK what trying more really :/

ATM I stay with OpenJDK which works properly (except RuneScape crash at the start of loading screen)

**Comment by DeicPro** [comment]
https://github.com/rsu-client/rsu-client/issues/144

**Comment by vb0** [comment]
For firefox I made work with browser.tabs.remote.autostart to false in about:config  - https://github.com/corbinlc/GNURootDebian/issues/234

I tried openjdk/IcedTea for a remote console for AsrockRack and while it does start and looks ok it doesn't connect to anything/all buttons are disabled (not sure if it just can't access the network or there's something else at play).

**Comment by DeicPro** [comment]
I installed Oracle JDK 8u172 ARM 64 (I'm using Redmi Note 5 Global 4 GB RAM/Android 8.1/sdm636 arm64-v8a) and when run RuneScape I got this:
```
Exception in thread "main" java.awt.HeadlessException                                             
        at java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:204)               
        at java.awt.Window.<init>(Window.java:536)                                                
        at java.awt.Frame.<init>(Frame.java:420) 
        at java.awt.Frame.<init>(Frame.java:385) 
        at app.appletviewer.a(appletviewer.java:507)                                              
        at jagexappletviewer.main(jagexappletviewer.java:34)
```

**Comment by pantumaca** [comment]
Thanks for posting the firefox fix, works fine.

What about libreoffice and its "proc not mounted" error? The "mount" command returns "proc on /proc"... Any fix?

**Comment by nakele** [comment]
Just leaving my feedback: 
Device Oneplus 3t stock Oreo 8.0

What i did was, launch Userland and followed the simple steps. Installed connectBot when prompted. Launched userLand just created environment. Typed password and then i typed those commands: 
su
apt-get update
apt-get upgrade
apt-get install lxde-core

At this point userLand crashed or closed every so often. I had to run dpkg --configure -a to continue and then again apt-get install lxde-core. This happened few times until lxde was completely installed. 

At this point i "exit" in order to go back to the normal user and then typed
export DISPLAY=127.0.0.1:0 PULSE_SERVER=tcp:127.0.0.1:4712

Than, you need to launch Xserver XSDL Android app before starting lxde. (Look in the PlayStore if you don't have installed)

And finally startlxde &

Go back to Xserver and after few seconds you will see the desktop appear.

I installed Firefox Debian counterpart but the page won't render as stated above. The page is actually loaded, i can use the web inspector to see the html but it's all white.

That's where I'm at right now

**Comment by DeicPro** [comment]
@ricecube did you got run RuneScape successfuly? Me & LordButtWiggle got the same errors.

**Comment by x1unix** [comment]
Same issue, android oreo, OnePlus 5

**Comment by deadport** [comment]
Same problem on HTC 10, Oreo

**Comment by GuyFromMars365** [comment]
Regarding @nakele comment is also discussed here: https://github.com/CypherpunkArmory/UserLAnd/issues/69

**Comment by stringyjosh94** [comment]
@DeicPro i got runelite running perfect, completely usable, setup UserLand with a debian filesystem, and ssh into it with connect bot, and install xserver android seperatley, reopen your ssh into connectbot type apt-update, apt-upgrade once these processes have finished install "apt-get install lxde-core" once installed i continued with installing a browser "apt-get install iceweasel" then install your java by using "apt-get install default-jre" once all setup export your display with "export DISPLAY=:0" then "startlxde &" (make sure xdsl server is running before using the export/start commands) then load up iceweasel you'll get a blank startup page you'll need to fix "browser.tabs.remote.autostart to false in about:config" once finished restart firefox (iceweasel) when reopened head to runelite.net download the java file from there site, once download head to your directory you instaled the java file to and open a terminal window at the directory, when terminal window has opened run "java -jar RuneLite.jar" ( this is a very rough guide of how i got it working, if you want to contact me feel free, i do have a copy of the osrs mobile beta and just prefer the keyboard/mouse method.

**Comment by DavidFahrenheit** [comment]
Bearing in mind most of this thread goes way over my head, is it possible for me to run Octave on Userland now? Thanks

**Comment by Thaytred** [comment]
@stringyjosh94 works like a charm on my Galaxy note 8. Cheers bud!

**Comment by DavidFahrenheit** [comment]
Got octave going with Connectbot, but getting a man in the middle warning. Anyone else get that?

**Comment by Bernhardino** [comment]
I installed Userland/Octave on my Galaxy S9 Oreo. It does run, but it seems, that a permanent internet connection is needed, although a big amount of Octave code was downloaded. If I remember properly,  Octave worked without internet a few years ago with my Galaxy S4 Lollipop. Can anybody confirm the current need of an internet connection? And why this? Thanks!
